src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -1240,18 +1161,25 @@ public new int[] Context
                 }
             }
         }
+
         private int[] _context;
+
         private int _preContext = 0;
+
         private int _postContext = 0;
 
+        private IContextTracker GetContextTracker() => (_preContext == 0 && _postContext == 0) ? _noContextTracker : new ContextTracker(_preContext, _postContext);
```

Comment:
We initialize _globalContextTracker in BeginProcessing so we could exclude _noContextTracker and GetContextTracker() at all.

Reply:
GetContextTracker is also used for every file, and I don't want to allocate a new tracker when there are nothing to track. I'm actually quite happy with the logic as it is.